frontend/src/components/review/StatusBadge.tsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import {
+  Badge,
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  Flex,
+} from "@chakra-ui/react";
+import { useMutation } from "@apollo/client";
+import { Icon } from "@chakra-ui/icon";
+import { MdCheck, MdTimer, MdArrowDropDown } from "react-icons/md";
+import {
+  getStatusVariant,
+  convertStatusTitleCase,
+} from "../../utils/StatusUtils";
+import {
+  UPDATE_STORY_TRANSLATION_CONTENT_STATUS,
+  UpdateStoryTranslationContentStatusResponse,
+} from "../../APIClients/mutations/StoryMutations";
+import { StoryLine } from "../translation/Autosave";
+
+export type StatusBadgeProps = {
+  translatedStoryLines: StoryLine[];
+  setTranslatedStoryLines: (storyLines: StoryLine[]) => void;
+  storyLine: StoryLine;
+  numApprovedLines: number;
+  setNumApprovedLines: (numLines: number) => void;
+};
+
+const StatusBadge = ({
+  translatedStoryLines,
+  setTranslatedStoryLines,
+  storyLine,
+  numApprovedLines,
+  setNumApprovedLines,
+}: StatusBadgeProps) => {
+  const [updateStatus] = useMutation<{
+    response: UpdateStoryTranslationContentStatusResponse;
+  }>(UPDATE_STORY_TRANSLATION_CONTENT_STATUS);
+
+  const handleStatusChange = async (
+    // storyLine: StoryLine,
+    newStatus: string,
+  ) => {
+    const result = updateStatus({
+      variables: {
+        storyTranslationContentId: storyLine.storyTranslationContentId!!,
+        status: newStatus,
+      },
+    });
+    if (result) {
+      const prevState = translatedStoryLines[storyLine.lineIndex].status;
+
+      const updatedStatusArray = [...translatedStoryLines!];
+      updatedStatusArray[storyLine.lineIndex].status = convertStatusTitleCase(
+        newStatus,
+      );
+
+      setTranslatedStoryLines!!(updatedStatusArray);
```

Comment:
are you _sure_ you need the `!!`s?
eg, does this work? `setTranslatedStoryLines(updatedStatusArray);`